libknet/compress_lz4.c

```diff
@@ -15,6 +15,8 @@
 #include "logging.h"
 #include "compress_model.h"
 
+#define ACCELERATION_DEFAULT 1
```

Comment:
I'll make this comment only once, but for each default, please add a comment from where the default is coming from.
For example:
ACCELERATION_DEFAULT 1 was found in man page X from library foo version 1.x
or
LZ4HC_DEFAULT was found in header file bla.h from library bar version 2.x

also, for all those default, we should have an #ifdef around them
#ifdef LZ4HC .....
#define KNET_COMPRESS_DEFAULT LZ4HC
#else 
#define KNET_COMPRESS_DEFAULT KNET_COMPRESS_UNKNOWN_DEFAULT
#endif